src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -70,16 +133,59 @@ fn parse_erc20(request: &pb::EthSignRequest) -> Option<([u8; 20], BigUint)> {
 }
 
 // fee: gas limit * gas price:
-fn parse_fee<'a>(request: &pb::EthSignRequest, params: &'a Params) -> Amount<'a> {
-    let gas_price = BigUint::from_bytes_be(&request.gas_price);
-    let gas_limit = BigUint::from_bytes_be(&request.gas_limit);
-    Amount {
-        unit: params.unit,
-        decimals: WEI_DECIMALS,
-        value: gas_price.mul(gas_limit),
+fn parse_fee<'a>(request: &Transaction<'a>, params: &'a Params) -> Amount<'a> {
+    let gas_limit = BigUint::from_bytes_be(request.gas_limit());
+    match request {
+        Transaction::Legacy(legacy) => {
+            let gas_price = BigUint::from_bytes_be(&legacy.gas_price);
+            Amount {
+                unit: params.unit,
+                decimals: WEI_DECIMALS,
+                value: gas_price.mul(gas_limit),
+            }
+        }
+        Transaction::Eip1559(eip1559) => {
+            let max_fee_per_gas = BigUint::from_bytes_be(&eip1559.max_fee_per_gas);
+            // We show the user the max possible fee, but the actual fee might be lower
+            Amount {
+                unit: params.unit,
+                decimals: WEI_DECIMALS,
+                value: max_fee_per_gas.mul(gas_limit),
+            }
+         }
     }
 }
 
+fn hash_legacy(chain_id: u64, recipient: [u8; 20], request: &pb::EthSignRequest) -> Result<[u8; 32], Error> {
+    let hash = super::sighash::compute_legacy(&super::sighash::ParamsLegacy {
+        nonce: &request.nonce,
+        gas_price: &request.gas_price,
+        gas_limit: &request.gas_limit,
+        recipient: &recipient,
+        value: &request.value,
+        data: &request.data,
+        chain_id,
+    })
+    .map_err(|_| Error::InvalidInput)?;
+    Ok(hash)
+}
+
+fn hash_eip1559(chain_id: u64, recipient: [u8; 20], request: &pb::EthSignEip1559Request) -> Result<[u8; 32], Error> {
```

Comment:
same here, drop the recipient arg in favor of request.recipient.

you can also drop the `chain_id` arg here as it is always present in `request.chain_id` (the same is not true in hash_legacy, as there the coin could also be specified with `request.coin` with a zero `request.chain_id`).